### PR TITLE
Front and back connect

### LINKED ISSUE #3: Add a textfield and button that demonstrates communication between frontend and backend
Our app needs a database! Unfortunately as of right now we only have communication between our frontend and backend.
(DB does not exist yet, so a blocker is waiting on ticket #2.)

- Create some type of input that shows you can pass JSON data back and forth between frontend and backend
- Allow this input to communicate between backend and database.
- Add a screenshot if you can

### PR DESCRIPTION
**Pull request by jordanjanakievski**
Fixes #3 

Creates buttons that enables the frontend to connect to backend APIs. Can send GET and POST requests.

Screenshot 2022-11-13 at 16 53 27

Screenshot 2022-11-13 at 16 59 59

Additional dependencies added 
In Server: body-parser
In Client: "react-scripts": "4.0.3" needed to be downgraded. Need feedback on this
"start": "WATCHPACK_POLLING=true react-scripts --openssl-legacy-provider start", required to solve client container bug

### CODE REVIEW

**Comment by kenneth-miura** [comment]
Downgrading react-scripts is fine imo

File: server/index.ts [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
 
 const express = require("express");
+const bodyParser = require("body-parser");
 
 const PORT = process.env.PORT || 3001;
 
 const app = express();
 
+// Handling CORS
+app.use((req, res, next) => {
```
Comment (kenneth-miura) [inline server/index.ts:11]: I'm pretty sure u don't actually need this, and the proxy in the client package.json just works

Comment (kenneth-miura) [reply server/index.ts:11]: Can u verify this + delete this if the proxy in `client` `package.json` works?

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
just a Q

File: server/index.ts (hunk 1 above)
Comment (jordanjanakievski) [reply server/index.ts:11]: I removed the cors packages in the newest commit. The `package,json` should no longer have cors

File: server/index.ts [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+		res.setHeader("Access-Control-Allow-Origin", origin);
+	}
+	res.header("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS");
+	res.header("Access-Control-Allow-Headers", "Content-Type, Authorization");
+	res.header("Access-Control-Allow-Credentials", true);
+	return next();
+});
+
+// Allow parsing of JSON data
+app.use(bodyParser.json());
```
Comment (helen-li) [inline server/index.ts:24]: We have express newer than 4.16 - I believe we don't need bodyParser and can just use `express.json()` here instead to allow parsing of JSON bodies in requests. Feel free to test everything without using bodyParser and remove the dependency if it's unnecessary.

**Review by helen-li — CHANGES_REQUESTED** [review]
Just one quick minor comment - otherwise everything looks good and works well! Tested in both browser and Postman

File: server/index.ts (hunk 2 above)
Comment (jordanjanakievski) [reply server/index.ts]: Thank you for the feedback. I removed the dependency and everything worked well.

**Review by helen-li — APPROVED** [review]
good to go for me!

File: server/index.ts (hunk 1 above)
Comment (kenneth-miura) [reply server/index.ts:9]: do u still need this function tho? If the proxy from frontend works.

Comment (jordanjanakievski) [reply server/index.ts]: Just tested and pushed a new commit. I removed the CORS handling since the server and client both connect without the need of it

**Review by kenneth-miura — APPROVED** [review]
lgtm!

**Pull request merged**